ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/launcher/TracingPropertySource.java

```diff
@@ -206,64 +203,42 @@ public void initialize() {
 		}
 	}
 
-	public TracingPropertySource(IPluginModelBase model, Properties masterOptions, Hashtable<String, String> template,
-			TracingBlock block) {
+	public TracingPropertySource(IPluginModelBase model, Map<String, String> masterOptions,
+			Map<String, String> template, TracingBlock block) {
 		fModel = model;
 		fMasterOptions = masterOptions;
 		fTemplate = template;
 		fBlock = block;
-		fValues = new Hashtable<>();
+		fValues = new HashMap<>();
 	}
 
 	public IPluginModelBase getModel() {
 		return fModel;
 	}
 
-	private Object[] getSortedKeys(int size) {
-		Object[] keyArray = new Object[size];
-		int i = 0;
-		for (Enumeration<String> keys = fTemplate.keys(); keys.hasMoreElements();) {
-			String key = keys.nextElement();
-			keyArray[i++] = key;
-		}
-		Arrays.sort(keyArray, this::compareKeys);
-		return keyArray;
-	}
-
-	private int compareKeys(Object o1, Object o2) {
-		String s1 = (String) o1;
-		String s2 = (String) o2;
-		// equal
-		return s1.compareTo(s2);
-	}
-
 	public void createContents(Composite parent, boolean enabled) {
-		fDescriptors = new Vector<>();
+		List<PropertyEditor> fDescriptors = new ArrayList<>();
```

Comment:
If it's not a field anymore then you should rename it to `descriptors`

Reply:
Good point.
But looking at it again, the list is only populated but never read, so effectively unused and can be removed.